From d4cb3243389bdc9e3d6d8d43669cd3c4cf9fd2e7 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 20 Jul 2021 12:35:47 +0300
Subject: [PATCH] [SW-45031] greco: add common dir

Added a blank file in arch/greco/common/.
This file should be used for moving functions which
are common between arc/arc_elf32/arm.

Change-Id: I4e0938aaee034031b7650405503e2aebe7274657
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/Makefile b/Makefile
index 42d0fb2..09ec9d3 100644
--- a/Makefile
+++ b/Makefile
@@ -41,11 +41,13 @@
 PREBOOT_LDLIBS = $(LDLIBS) -lfdt
 
 PREBOOT_SRCS_C := $(shell ls lib/common/*.c)
+PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/*.c)
 PREBOOT_SRCS_C += $(shell ls lib/preboot/*.c)
 PREBOOT_SRCS_C += $(shell ls lib/common/lzo/*.c)
 PREBOOT_SRCS_C += $(shell ls app/preboot/*.c)
 PREBOOT_SRCS_C := $(filter-out lib/preboot/thermal_v2.c, $(PREBOOT_SRCS_C))
 PPBOOT_SRCS_C := $(shell ls lib/common/*.c)
+PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/*.c)
 PPBOOT_SRCS_C += $(shell ls app/ppboot/*.c)
 
 include $(HL_DEV_ARCH)/Makefile
diff --git a/app/recovery/Makefile b/app/recovery/Makefile
index d30c09d..cceb955 100644
--- a/app/recovery/Makefile
+++ b/app/recovery/Makefile
@@ -23,6 +23,7 @@
 
 RECOVERY_SRCS_C := $(shell ls *.c)
 RECOVERY_SRCS_C += $(shell ls ../../lib/common/*.c)
+RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/../common/*.c)
 RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/common.c)
 RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/$(HL_TARGET).c)
 RECOVERY_APP_S := $(shell ls crt0_recovery.S)
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
new file mode 100644
index 0000000..599b1bc
--- /dev/null
+++ b/arch/greco/common/common.c
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * (C) Copyright 2021 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ */
+
+/* TODO: Commons functions in arc, arc_el32, arm to be moved here. */
